### PR TITLE
Hds 1794 header docs

### PR DESCRIPTION
**Pull request by harriplappalainen**
## Description

- Document Header components as they are now
  - Extensive docs for sub components props as they can't be seen in storybook
  - NavigationSearch and NavigationLanguageSelector components are still up in the air on how they're going to end up
- Small fixes to Header components JSDocs
- Change prop names to href instead of url. It's semantically correct and is more consistent with HDS
- Add prop tables for Footer components as well
- Small fixes to Footer components JSDocs

## Related Issue

https://helsinkisolutionoffice.atlassian.net/browse/HDS-1794

## How Has This Been Tested?
Locally, running tests
[Demo](https://city-of-helsinki.github.io/hds-demo/header-docs)

To do:
- [x] Docs about ActionBarItem

### CODE REVIEW

File: site/src/docs/components/header/index.mdx [hunk 1]
```diff
@@ -22,56 +22,232 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
...
+          />,
+        ]}
+      />
+      <Header.NavigationLink href="#" label="Link 3" />
+    </Header.NavigationMenu>
+  </Header>
+</PlaygroundPreview>
 
-#### With navigation bar
+#### With universal bar
```
Comment (laurakarhu) [inline site/src/docs/components/header/index.mdx:180]: Universal bar is aimed for frequenlty visited and useful links, e.g. feedback and news.

File: site/src/docs/components/header/index.mdx [hunk 2]
```diff
@@ -22,56 +22,232 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
...
+      titleHref="https://hel.fi"
+      titleHref="https://hel.fi"
+      logoAriaLabel="Service logo"
+      logoHref="https://hel.fi"
+      menuButtonAriaLabel="Menu"
+    />
+  </Header>
+</PlaygroundPreview>
 
 #### With navigation language selector
```
Comment (laurakarhu) [inline site/src/docs/components/header/index.mdx:204]: Language selector changes the language for the whole service. It is recommended to offer full Finnish, Swedish and English versions. Other languages can be added based on the need and target audience of the service. Smaller subset of content can be separated from fully translated content.

File: site/src/docs/components/header/index.mdx [hunk 3]
```diff
@@ -22,56 +22,232 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
...
 
 - Using the Header.NavigationLanguageSelector is optional. Use the navigation language selector to define site languages. 
-- Header has <code>onDidChangeLanguage</code> function, where user can define a callback function which gets the selected language key (value) as parameter.
+- The Header component has `onDidChangeLanguage` prop function, where user can define a callback function which gets the selected language key (value) as parameter.
 - See example for Header.NavigationLanguageSelector <Link openInNewTab openInNewTabAriaLabel="Opens in a new tab" href="/storybook/react/?path=/story/components-header--minimal-with-localization">in Storybook</Link>
+- On smaller screens (768px and down) the Header.NavigationLanguageSelector moves under the Header.ActionBar.
 
 #### With skip link
```
Comment (laurakarhu) [inline site/src/docs/components/header/index.mdx:211]: Use the skip link component to help keyboard-only users skip to the main content on a page.

File: site/src/docs/components/header/index.mdx [hunk 4]
```diff
@@ -22,56 +22,232 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
...
 
 - Using the Header.SkipLink component is optional. Use the skip link component to help keyboard-only users skip to the main content on a page.
-- See example for Header.SkipLink <Link openInNewTab openInNewTabAriaLabel="Opens in a new tab" href="/storybook/react/?path=/story/components-header--with-skip-link">in Storybook</Link>
-- The Header.SkipLink supports ariaLabel which gives possibility to define more informative text for screen readers
+- The Header.SkipLink supports `ariaLabel` prop which gives possibility to define more informative text for screen readers.
+
```
Comment (laurakarhu) [inline site/src/docs/components/header/index.mdx:214]: It is recommneded to have the skip link, especially if there is any other component in addition to the action bar.

**Review by laurakarhu — COMMENTED** [review]
Good job, I had couple of suggestions to enhance the documentation.

Also, still thinking of the order:should it be the same as in the header or should it be mandatory first as it is now?

**Comment by harriplappalainen** [comment]
@laurakarhu Thank you for comments! I fixed them and added some more good stuff. Check it out.
Additional fixes:
- Old Navigation component has link to new Header docs
- Added StatusLabelTooltip
- Info about ActionBar's `titleStyle` prop usage
- Info about UniversalBar's primaryLink usage

File: site/src/docs/components/header/index.mdx [hunk 5]
```diff
@@ -17,61 +17,289 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
...
-- **It is strongly recommended to always include the header component in your service.**
+- **It is strongly recommended to always include the header component on every page in your service.**
 - The Header component is a wrapper for the HDS header navigation components. This means you should put all the other header elements inside this component.
 - The only mandatory component inside the header is HeaderActionBar. Other components may be used if needed.
-- HeaderNavigationBar is coming soon
-- HeaderActionBar is coming soon
 - The inner components should **always** be in this order: Header.SkipLink, Header.UniversalBar, HeaderActionBar and HeaderNavigationBar.
+- Try to keep the order of actions (language select, search, log in) consistent in navigation. Do not change the default order without a good reason.
 
 ### Variations
```
Comment (laurakarhu) [inline site/src/docs/components/header/index.mdx:106]: Should we mention here also that the order to make it clear?
Maybe also a list with anchor links


Variations are presented in the same order than they appear in the actual header.
- With skip link
- With universal bar
- With action bar
- With navigation bar
- With navigation language selector
- With search

File: site/src/docs/components/header/index.mdx [hunk 6]
```diff
@@ -17,61 +17,289 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
...
 ### Principles
 
-- **It is strongly recommended to always include the header component in your service.**
+- **It is strongly recommended to always include the header component on every page in your service.**
 - The Header component is a wrapper for the HDS header navigation components. This means you should put all the other header elements inside this component.
 - The only mandatory component inside the header is HeaderActionBar. Other components may be used if needed.
-- HeaderNavigationBar is coming soon
-- HeaderActionBar is coming soon
 - The inner components should **always** be in this order: Header.SkipLink, Header.UniversalBar, HeaderActionBar and HeaderNavigationBar.
+- Try to keep the order of actions (language select, search, log in) consistent in navigation. Do not change the default order without a good reason.
```
Comment (laurakarhu) [inline site/src/docs/components/header/index.mdx:104]: To better understand how navigation elements work and how they should be combined, see also Navigation pattern (link)

File: site/src/docs/components/header/accessibility.mdx [hunk 7]
```diff
@@ -16,3 +16,4 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
...
 - **When designing a menu link hierarchy, always think about keyboard and screen reader users.** First, make sure top-level menu labels are clear, and submenu items contextually fit under it. Next, make sure the menu option order is logical and reasonable. Remember that keyboard users need to manually navigate to each element and thus placing a commonly used menu option last is not optimal.
 - Landmarks are provided by the wrapping `<header>` element, `<nav>` element in the Header.NavigationMenu component and with `role="search"` attribute on the Header.NavigationSearch component. In case you want more landmarks, the Header.UniversalBar, Header.ActionBar and Header.NavigationSearch components support `role` prop which should be used together with a descriptive `ariaLabel` prop. Just remember that landmarks should be used sparingly in order to avoid "noise" for screen readers and making the overall page layout difficult to understand.
+- Header.NavigationMenu's dropdowns open only by clicking them. They don't open from hovering with the mouse as it can be difficult to navigate for people with motor and seeing challenges.
```
Comment (laurakarhu) [inline site/src/docs/components/header/accessibility.mdx:19]: Even though HDS header supports larger needs, only use the elements you need. Avoid clutter.

**Review by laurakarhu — COMMENTED** [review]
Looking really good, just couple of suggestions to add still, links and notes.

**Review by laurakarhu — APPROVED** [review]
Hyvää pyysin, mutta priimaa pukkaa! Kiitos!

**Review by mrTuomoK — APPROVED** [review]
I don't see anything suspicious here 😄 a Goood good 👍

**Pull request merged**